server/src/main/java/keywhiz/service/daos/SecretSeriesDAO.java

```diff
@@ -306,6 +317,20 @@ public void deleteSecretSeriesById(long id) {
     });
   }
 
+  public void renameSecretSeriesById(long secretId, String name) {
+    dslContext.update(SECRETS)
+        .set(SECRETS.NAME, name)
+        .where(SECRETS.ID.eq(secretId))
+        .execute();
+  }
+
+  public void updateSecretSeriesContentById(long secretId, long secretContentId) {
```

Comment:
The existing `setCurrentVersion` endpoint will actually handle this already, with some additional checks (for instance, making sure that this `secretContentId` was linked to this `secretId`).